### ISSUE TITLE
Install failing in postinstall script

### ISSUE THREAD

**Issue by hanzlamateen** [versatica/mediasoup#854]
## Bug Report

### Your environment

- Operating system: Ubuntu 20.04.4 LTS
- Node version: v17.5.0
- npm version: 8.7.0
- gcc/clang version: 9.4.0 (Ubuntu 9.4.0-1ubuntu1~20.04.1)
- mediasoup version: Tried both 3.9.6 & 3.10.0
- mediasoup-client version: Tried both 3.6.51 & 3.6.53


### Issue description

I am getting issue when running `npm install` locally as well as in CI. Here is the link of it failing in our github action [link](https://github.com/XRFoundation/XREngine/runs/7173489800?check_suite_focus=true)

Below is the log of error on my local console where running in docker build:

```
=> ERROR [21/24] RUN npm install --loglevel notice --legacy-peer-deps                                                                                                                                                                 920.2s
------                                                                                                                                                                                                                                        
 > [21/24] RUN npm install --loglevel notice --legacy-peer-deps:                                                                                                                                                                              
#25 601.8 npm WARN deprecated rollup-plugin-inject@3.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-inject.                                                                                     
#25 627.5 npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.                                                                                               
#25 645.6 npm WARN deprecated multer@1.4.2: Multer 1.x is affected by CVE-2022-24434. This is fixed in v1.4.4-lts.1 which drops support for versions of Node.js before 6. Please upgrade to at least Node.js 6 and version 1.4.4-lts.1 of Multer. If you need support for older versions of Node.js, we are open to accepting patches that would fix the CVE on the main 1.x release line, whilst maintaining compatibility with Node.js 0.10.                                              
#25 646.6 npm WARN deprecated feathers-sequelize@6.3.2: Upgrade to feathers-sequelize 6.3.3 or later for important security and dependency updates
#25 650.1 npm WARN deprecated feathers-logger@0.3.2: This module is no longer maintained. See https://github.com/feathersjs-ecosystem/feathers-logger#migrating for more information on using a standard logger
#25 662.2 npm WARN deprecated emailjs-com@3.2.0: The SDK name changed to @emailjs/browser
#25 666.5 npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
#25 667.2 npm WARN deprecated har-validator@5.1.5: this library is no longer supported
#25 672.4 npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
#25 675.2 npm WARN deprecated read-package-tree@5.3.1: The functionality that this package provided is now in @npmcli/arborist
#25 678.8 npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
#25 696.0 npm WARN deprecated @types/pino@7.0.5: This is a stub types definition. pino provides its own type definitions, so you do not need this installed.
#25 708.7 npm WARN deprecated uuid@3.3.2: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
#25 920.0 npm ERR! code 1
#25 920.0 npm ERR! path /app/node_modules/mediasoup
#25 920.0 npm ERR! command failed
#25 920.0 npm ERR! command sh -c /tmp/postinstall656946593.sh
#25 920.0 npm ERR! npm-scripts.js [INFO] running task "postinstall"
#25 920.0 npm ERR! npm-scripts.js [INFO] executing command: node npm-scripts.js worker:build
#25 920.0 npm ERR! npm-scripts.js [INFO] running task "worker:build"
#25 920.0 npm ERR! npm-scripts.js [INFO] executing command: make -C worker
#25 920.0 npm ERR! make: Entering directory '/app/node_modules/mediasoup/worker'
#25 920.0 npm ERR! # Updated pip and setuptools are needed for meson
#25 920.0 npm ERR! # `--system` is not present everywhere and is only needed as workaround for
#25 920.0 npm ERR! # Debian-specific issue (copied from
#25 920.0 npm ERR! # https://github.com/gluster/gstatus/pull/33), fallback to command without
#25 920.0 npm ERR! # `--system` if the first one fails.
#25 920.0 npm ERR! /usr/bin/python3 -m pip install --system --target=/app/node_modules/mediasoup/worker/out/pip pip setuptools || \
#25 920.0 npm ERR! 	/usr/bin/python3 -m pip install --target=/app/node_modules/mediasoup/worker/out/pip pip setuptools || \
#25 920.0 npm ERR! 	echo "Installation failed, likely because PIP is unavailable, if you are on Debian/Ubuntu or derivative please install the python3-pip package"
#25 920.0 npm ERR! Collecting pip
#25 920.0 npm ERR!   Downloading https://files.pythonhosted.org/packages/96/2f/caec18213f6a67852f6997fb0673ae08d2e93d1b81573edb93ba4ef06970/pip-22.1.2-py3-none-any.whl (2.1MB)
#25 920.0 npm ERR! Collecting setuptools
#25 920.0 npm ERR!   Downloading https://files.pythonhosted.org/packages/ae/7f/6d816941769a7783be4258dd35e28bbf1a64bb36b1b7e0c773eff07fb0a8/setuptools-63.1.0-py3-none-any.whl (1.2MB)
#25 920.0 npm ERR! Installing collected packages: pip, setuptools
#25 920.0 npm ERR! Successfully installed pip-22.1.2 setuptools-63.1.0
#25 920.0 npm ERR! # Install `meson` and `ninja` using `pip` into custom location, so we don't
#25 920.0 npm ERR! # depend on system-wide installation.
#25 920.0 npm ERR! /usr/bin/python3 -m pip install --upgrade --target=/app/node_modules/mediasoup/worker/out/pip  meson ninja
#25 920.0 npm ERR! Collecting meson
#25 920.0 npm ERR!   Downloading meson-0.63.0-py3-none-any.whl (884 kB)
#25 920.0 npm ERR!      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 885.0/885.0 kB 1.8 MB/s eta 0:00:00
#25 920.0 npm ERR! Collecting ninja
#25 920.0 npm ERR!   Downloading ninja-1.10.2.3-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (108 kB)
#25 920.0 npm ERR!      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 108.1/108.1 kB 82.2 kB/s eta 0:00:00
#25 920.0 npm ERR! Installing collected packages: ninja, meson
#25 920.0 npm ERR! Successfully installed meson-0.63.0 ninja-1.10.2.3
#25 920.0 npm ERR! /app/node_modules/mediasoup/worker/out/pip/bin/meson setup \
#25 920.0 npm ERR! 	--prefix /app/node_modules/mediasoup/worker/out/Release \
#25 920.0 npm ERR! 	--bindir '' \
#25 920.0 npm ERR! 	--libdir '' \
#25 920.0 npm ERR! 	--buildtype release \
#25 920.0 npm ERR! 	-Db_ndebug=true \
#25 920.0 npm ERR! 	-Db_pie=true \
#25 920.0 npm ERR! 	-Db_staticpic=true \
#25 920.0 npm ERR! 	--reconfigure \
#25 920.0 npm ERR! 	"" \
#25 920.0 npm ERR! 	/app/node_modules/mediasoup/worker/out/Release/build || \
#25 920.0 npm ERR! 	/app/node_modules/mediasoup/worker/out/pip/bin/meson setup \
#25 920.0 npm ERR! 		--prefix /app/node_modules/mediasoup/worker/out/Release \
#25 920.0 npm ERR! 		--bindir '' \
#25 920.0 npm ERR! 		--libdir '' \
#25 920.0 npm ERR! 		--buildtype release \
#25 920.0 npm ERR! 		-Db_ndebug=true \
#25 920.0 npm ERR! 		-Db_pie=true \
#25 920.0 npm ERR! 		-Db_staticpic=true \
#25 920.0 npm ERR! 		"" \
#25 920.0 npm ERR! 		/app/node_modules/mediasoup/worker/out/Release/build
#25 920.0 npm ERR! The Meson build system
#25 920.0 npm ERR! Version: 0.63.0
#25 920.0 npm ERR! Source dir: /app/node_modules/mediasoup/worker
#25 920.0 npm ERR! Build dir: /app/node_modules/mediasoup/worker/out/Release/build
#25 920.0 npm ERR! Build type: native build
#25 920.0 npm ERR! Project name: mediasoup-worker
#25 920.0 npm ERR! Project version: undefined
#25 920.0 npm ERR! C compiler for the host machine: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! C linker for the host machine: cc ld.bfd 2.31.1
#25 920.0 npm ERR! C++ compiler for the host machine: c++ (gcc 8.3.0 "c++ (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! C++ linker for the host machine: c++ ld.bfd 2.31.1
#25 920.0 npm ERR! Host machine cpu family: x86_64
#25 920.0 npm ERR! Host machine cpu: x86_64
#25 920.0 npm ERR! Downloading openssl source from https://www.openssl.org/source/openssl-3.0.2.tar.gz
#25 920.0 npm ERR! Download size: 15038141
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! Downloading openssl patch from https://wrapdb.mesonbuild.com/v2/openssl_3.0.2-1/get_patch
#25 920.0 npm ERR! Download size: 11837869
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject openssl 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! openssl| Project name: openssl
#25 920.0 npm ERR! openssl| Project version: 3.0.2
#25 920.0 npm ERR! openssl| C compiler for the host machine: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! openssl| C linker for the host machine: cc ld.bfd 2.31.1
#25 920.0 npm ERR! openssl| Run-time dependency threads found: YES
#25 920.0 npm ERR! openssl| Program as found: YES (/usr/bin/as)
#25 920.0 npm ERR! openssl| Message: OpenSSL is configured with ASM support
#25 920.0 npm ERR! openssl| Library dl found: YES
#25 920.0 npm ERR! openssl| Build targets in project: 3
#25 920.0 npm ERR! openssl| Subproject openssl finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading nlohmann_json source from https://github.com/nlohmann/json/releases/download/v3.10.5/include.zip
#25 920.0 npm ERR! Download size: 260679
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject nlohmann_json 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! nlohmann_json| Project name: nlohmann_json
#25 920.0 npm ERR! nlohmann_json| Project version: 3.10.5
#25 920.0 npm ERR! nlohmann_json| C++ compiler for the host machine: c++ (gcc 8.3.0 "c++ (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! nlohmann_json| C++ linker for the host machine: c++ ld.bfd 2.31.1
#25 920.0 npm ERR! nlohmann_json| Build targets in project: 3
#25 920.0 npm ERR! nlohmann_json| Subproject nlohmann_json finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading libuv source from https://dist.libuv.org/dist/v1.43.0/libuv-v1.43.0.tar.gz
#25 920.0 npm ERR! Download size: 1288663
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! Downloading libuv patch from https://wrapdb.mesonbuild.com/v2/libuv_1.43.0-1/get_patch
#25 920.0 npm ERR! Download size: 5153
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject libuv 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! libuv| Project name: libuv
#25 920.0 npm ERR! libuv| Project version: 1.43.0
#25 920.0 npm ERR! libuv| C compiler for the host machine: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! libuv| C linker for the host machine: cc ld.bfd 2.31.1
#25 920.0 npm ERR! libuv| Fetching value of define "__QNX__" :
#25 920.0 npm ERR! libuv| Fetching value of define "_AIX" :
#25 920.0 npm ERR! libuv| Message: System: linux
#25 920.0 npm ERR! libuv| Compiler for C supports arguments -fno-strict-aliasing: YES
#25 920.0 npm ERR! libuv| Dependency threads found: YES unknown (cached)
#25 920.0 npm ERR! libuv| Library dl found: YES
#25 920.0 npm ERR! libuv| Library rt found: YES
#25 920.0 npm ERR! libuv| Build targets in project: 4
#25 920.0 npm ERR! libuv| Subproject libuv finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading libsrtp2 source from https://github.com/cisco/libsrtp/archive/refs/tags/v2.4.2.zip
#25 920.0 npm ERR! Download size: 1286149
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject libsrtp2 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! libsrtp2| Project name: libsrtp2
#25 920.0 npm ERR! libsrtp2| Project version: 2.4.2
#25 920.0 npm ERR! libsrtp2| C compiler for the host machine: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! libsrtp2| C linker for the host machine: cc ld.bfd 2.31.1
#25 920.0 npm ERR! libsrtp2| Has header "arpa/inet.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "byteswap.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "inttypes.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "machine/types.h" : NO
#25 920.0 npm ERR! libsrtp2| Has header "netinet/in.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "stdint.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "stdlib.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "sys/int_types.h" : NO
#25 920.0 npm ERR! libsrtp2| Has header "sys/socket.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "sys/types.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "sys/uio.h" : YES
#25 920.0 npm ERR! libsrtp2| Has header "unistd.h" : YES
#25 920.0 npm ERR! libsrtp2| Checking for function "sigaction" : YES
#25 920.0 npm ERR! libsrtp2| Checking for function "inet_aton" : YES
#25 920.0 npm ERR! libsrtp2| Checking for function "usleep" : YES
#25 920.0 npm ERR! libsrtp2| Checking for function "socket" : YES
#25 920.0 npm ERR! libsrtp2| Has header "stdint.h" : YES (cached)
#25 920.0 npm ERR! libsrtp2| Checking for type "size_t" : YES
#25 920.0 npm ERR! libsrtp2| Checking for type "unsigned long" : YES
#25 920.0 npm ERR! libsrtp2| Checking for size of "unsigned long" : 8
#25 920.0 npm ERR! libsrtp2| Checking for type "unsigned long long" : YES
#25 920.0 npm ERR! libsrtp2| Checking for size of "unsigned long long" : 8
#25 920.0 npm ERR! libsrtp2| Checking if "inline keyword check" compiles: YES
#25 920.0 npm ERR! libsrtp2| Dependency openssl from subproject subprojects/openssl-3.0.2 found: YES 3.0.2
#25 920.0 npm ERR! libsrtp2| Configuring config.h using configuration
#25 920.0 npm ERR! libsrtp2| Compiler for C supports arguments -Wstrict-prototypes: YES
#25 920.0 npm ERR! libsrtp2| Compiler for C supports arguments -funroll-loops: YES
#25 920.0 npm ERR! libsrtp2| Program doxygen found: NO
#25 920.0 npm ERR! libsrtp2| Build targets in project: 5
#25 920.0 npm ERR! libsrtp2| Subproject libsrtp2 finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading usrsctp source from https://github.com/sctplab/usrsctp/archive/4e06feb01cadcd127d119486b98a4bd3d64aa1e7.zip
#25 920.0 npm ERR! Download size: 948210
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject usrsctp 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! usrsctp| Project name: usrsctplib
#25 920.0 npm ERR! usrsctp| Project version: 0.9.5.0
#25 920.0 npm ERR! usrsctp| C compiler for the host machine: cc (gcc 8.3.0 "cc (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! usrsctp| C linker for the host machine: cc ld.bfd 2.31.1
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -pedantic: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wfloat-equal: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wshadow: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wpointer-arith: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Winit-self: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wno-unused-function: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wno-unused-parameter: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wno-unreachable-code: YES
#25 920.0 npm ERR! usrsctp| Compiler for C supports arguments -Wstrict-prototypes: YES (cached)
#25 920.0 npm ERR! usrsctp| Dependency threads found: YES unknown (cached)
#25 920.0 npm ERR! usrsctp| Has header "sys/queue.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "sys/socket.h" : YES (cached)
#25 920.0 npm ERR! usrsctp| Has header "linux/if_addr.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "linux/rtnetlink.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "sys/types.h" : YES (cached)
#25 920.0 npm ERR! usrsctp| Has header "netinet/in.h" : YES (cached)
#25 920.0 npm ERR! usrsctp| Has header "netinet/ip.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "netinet/ip_icmp.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "net/route.h" : YES
#25 920.0 npm ERR! usrsctp| Has header "stdatomic.h" : YES
#25 920.0 npm ERR! usrsctp| Checking whether type "struct sockaddr" has member "sa_len" : NO
#25 920.0 npm ERR! usrsctp| Checking whether type "struct sockaddr_in" has member "sin_len" : NO
#25 920.0 npm ERR! usrsctp| Checking whether type "struct sockaddr_in6" has member "sin6_len" : NO
#25 920.0 npm ERR! usrsctp| Checking whether type "struct sockaddr_conn" has member "sconn_len" : NO
#25 920.0 npm ERR! usrsctp| Build targets in project: 6
#25 920.0 npm ERR! usrsctp| Subproject usrsctp finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading abseil-cpp source from https://github.com/abseil/abseil-cpp/archive/20211102.0.tar.gz
#25 920.0 npm ERR! Download size: 1884080
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! Downloading abseil-cpp patch from https://wrapdb.mesonbuild.com/v2/abseil-cpp_20211102.0-2/get_patch
#25 920.0 npm ERR! Download size: 3404
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject abseil-cpp 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! abseil-cpp| Project name: abseil-cpp
#25 920.0 npm ERR! abseil-cpp| Project version: 20211102.0
#25 920.0 npm ERR! abseil-cpp| C++ compiler for the host machine: c++ (gcc 8.3.0 "c++ (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! abseil-cpp| C++ linker for the host machine: c++ ld.bfd 2.31.1
#25 920.0 npm ERR! abseil-cpp| Compiler for C++ supports arguments -Wno-sign-compare: YES
#25 920.0 npm ERR! abseil-cpp| Compiler for C++ supports arguments -maes: YES
#25 920.0 npm ERR! abseil-cpp| Compiler for C++ supports arguments -msse4.1: YES
#25 920.0 npm ERR! abseil-cpp| Checking if "GCC atomic builtins" : links: YES
#25 920.0 npm ERR! abseil-cpp| Dependency threads found: YES unknown (cached)
#25 920.0 npm ERR! abseil-cpp| Did not find pkg-config by name 'pkg-config'
#25 920.0 npm ERR! abseil-cpp| Found Pkg-config: NO
#25 920.0 npm ERR! abseil-cpp| Did not find CMake 'cmake'
#25 920.0 npm ERR! abseil-cpp| Found CMake: NO
#25 920.0 npm ERR! abseil-cpp| Run-time dependency corefoundation found: NO (tried pkgconfig and cmake)
#25 920.0 npm ERR! abseil-cpp| Build targets in project: 19
#25 920.0 npm ERR! abseil-cpp| Subproject abseil-cpp finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Downloading catch2 source from https://github.com/catchorg/Catch2/archive/v2.13.7.zip
#25 920.0 npm ERR! Download size: 846562
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! Downloading catch2 patch from https://wrapdb.mesonbuild.com/v2/catch2_2.13.7-1/get_patch
#25 920.0 npm ERR! Download size: 1133
#25 920.0 npm ERR! Downloading: ..........
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Executing subproject catch2 
#25 920.0 npm ERR! 
#25 920.0 npm ERR! catch2| Project name: catch2
#25 920.0 npm ERR! catch2| Project version: 2.13.7
#25 920.0 npm ERR! catch2| C++ compiler for the host machine: c++ (gcc 8.3.0 "c++ (Debian 8.3.0-6) 8.3.0")
#25 920.0 npm ERR! catch2| C++ linker for the host machine: c++ ld.bfd 2.31.1
#25 920.0 npm ERR! catch2| Build targets in project: 19
#25 920.0 npm ERR! catch2| Subproject catch2 finished.
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Build targets in project: 24
#25 920.0 npm ERR! 
#25 920.0 npm ERR! mediasoup-worker undefined
#25 920.0 npm ERR! 
#25 920.0 npm ERR!   Subprojects
#25 920.0 npm ERR!     abseil-cpp   : YES
#25 920.0 npm ERR!     catch2       : YES
#25 920.0 npm ERR!     libsrtp2     : YES
#25 920.0 npm ERR!     libuv        : YES
#25 920.0 npm ERR!     nlohmann_json: YES
#25 920.0 npm ERR!     openssl      : YES
#25 920.0 npm ERR!     usrsctp      : YES
#25 920.0 npm ERR! 
#25 920.0 npm ERR!   User defined options
#25 920.0 npm ERR!     bindir       : 
#25 920.0 npm ERR!     buildtype    : release
#25 920.0 npm ERR!     libdir       : 
#25 920.0 npm ERR!     prefix       : /app/node_modules/mediasoup/worker/out/Release
#25 920.0 npm ERR!     b_ndebug     : true
#25 920.0 npm ERR!     b_pie        : true
#25 920.0 npm ERR!     b_staticpic  : true
#25 920.0 npm ERR! 
#25 920.0 npm ERR! Found ninja-1.10.2.git.kitware.jobserver-1 at /app/node_modules/mediasoup/worker/out/pip/bin/ninja
#25 920.0 npm ERR! /app/node_modules/mediasoup/worker/out/pip/bin/meson compile -C /app/node_modules/mediasoup/worker/out/Release/build -j 8 mediasoup-worker
#25 920.0 npm ERR! ninja: Entering directory `/app/node_modules/mediasoup/worker/out/Release/build'
#25 920.0 npm ERR! [1/1322] Compiling C object subprojects/openssl-3.0.2/libcrypto.a.p/crypto_aes_aes_ofb.c.o
#25 920.0 npm ERR! [2/1322] Compiling C object subprojects/openssl-3.0.2/libcrypto.a.p/crypto_aes_aes_cfb.c.o

.....

#25 920.0 npm ERR! [999/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_uv-common.c.o
#25 920.0 npm ERR! [1000/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o
#25 920.0 npm ERR! FAILED: subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o 
#25 920.0 npm ERR! cc -Isubprojects/libuv-v1.43.0/libuv.a.p -Isubprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0 -I../../../subprojects/libuv-v1.43.0/include -I../../../subprojects/libuv-v1.43.0/src -fdiagnostics-color=always -DNDEBUG -std=c89 -O3 -fPIC -pthread -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112 -MD -MQ subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -MF subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o.d -o subprojects/libuv-v1.43.0/libuv.a.p/src_unix_stream.c.o -c ../../../subprojects/libuv-v1.43.0/src/unix/stream.c
#25 920.0 npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c: In function ‘uv__write’:
#25 920.0 npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: error: C++ style comments are not allowed in ISO C90
#25 920.0 npm ERR!    // XXX(jwn): this must call uv__stream_flush_write_queue(stream, n) here, since we won't generate any more events
#25 920.0 npm ERR!    ^
#25 920.0 npm ERR! ../../../subprojects/libuv-v1.43.0/src/unix/stream.c:929:3: error: (this will be reported only once per input file)
#25 920.0 npm ERR! [1001/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_process.c.o
#25 920.0 npm ERR! [1002/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_tcp.c.o
#25 920.0 npm ERR! [1003/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_tty.c.o
#25 920.0 npm ERR! [1004/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_thread.c.o
#25 920.0 npm ERR! [1005/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_signal.c.o
#25 920.0 npm ERR! [1006/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_fs.c.o
#25 920.0 npm ERR! [1007/1322] Compiling C object subprojects/libuv-v1.43.0/libuv.a.p/src_unix_udp.c.o
#25 920.0 npm ERR! ninja: build stopped: subcommand failed.
#25 920.0 npm ERR! make: Leaving directory '/app/node_modules/mediasoup/worker'
#25 920.0 npm ERR! WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#25 920.0 npm ERR! Directory does not contain a valid build tree:
#25 920.0 npm ERR! /app/node_modules/mediasoup/worker/out/Release/build
#25 920.0 npm ERR! make: *** [Makefile:168: mediasoup-worker] Error 1
#25 920.0 
#25 920.0 npm ERR! A complete log of this run can be found in:
#25 920.0 npm ERR!     /root/.npm/_logs/2022-07-04T05_50_50_261Z-debug-0.log
------
executor failed running [/bin/sh -c npm install --loglevel notice --legacy-peer-deps]: exit code: 1
```

**Comment by nazar-pc** [comment]
We test mediasoup in CI on all major platforms and it doesn't fail there with this error, hence must be something with your project.

With these kinds of questions create thread on the forum: https://mediasoup.discourse.group/

**Comment by mizoRC** [comment]
We are getting the same problem. 
Base image: node:16.15.0-bullseye-slim
Node v14.19.2
Npm 6.14.17
Gcc Apple clang version 13.1.6 (clang-1316.0.21.2)
Mediasoup version 3.9.2

We make a build on friday with exactly the same code and versions, and it works.

```
#8 97.04 npm ERR! [727/992] Compiling C object subprojects/libuv-v1.42.0/libuv.a.p/src_unix_random-devurandom.c.o
#8 97.04 npm ERR! [728/992] Compiling C object subprojects/libuv-v1.42.0/libuv.a.p/src_unix_stream.c.o
#8 97.04 npm ERR! FAILED: subprojects/libuv-v1.42.0/libuv.a.p/src_unix_stream.c.o 
#8 97.04 npm ERR! cc -Isubprojects/libuv-v1.42.0/libuv.a.p -Isubprojects/libuv-v1.42.0 -I../../subprojects/libuv-v1.42.0 -I../../subprojects/libuv-v1.42.0/include -I../../subprojects/libuv-v1.42.0/src -fdiagnostics-color=always -DNDEBUG -std=c89 -O3 -fPIC -pthread -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112 -MD -MQ subprojects/libuv-v1.42.0/libuv.a.p/src_unix_stream.c.o -MF subprojects/libuv-v1.42.0/libuv.a.p/src_unix_stream.c.o.d -o subprojects/libuv-v1.42.0/libuv.a.p/src_unix_stream.c.o -c ../../subprojects/libuv-v1.42.0/src/unix/stream.c
#8 97.04 npm ERR! ../../subprojects/libuv-v1.42.0/src/unix/stream.c: In function ‘uv__write’:
#8 97.04 npm ERR! ../../subprojects/libuv-v1.42.0/src/unix/stream.c:929:3: error: C++ style comments are not allowed in ISO C90
#8 97.04 npm ERR!   929 |   // XXX(jwn): this must call uv__stream_flush_write_queue(stream, n) here, since we won't generate any more events
#8 97.04 npm ERR!       |   ^
#8 97.04 npm ERR! ../../subprojects/libuv-v1.42.0/src/unix/stream.c:929:3: note: (this will be reported only once per input file)
#8 97.04 npm ERR! [729/992] Compiling C object subprojects/libuv-v1.42.0/libuv.a.p/src_unix_core.c.o
#8 97.04 npm ERR! ninja: build stopped: subcommand failed.
#8 97.04 npm ERR! make: Leaving directory '/app/node_modules/mediasoup/worker'
#8 97.04 npm ERR! WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 97.04 npm ERR! Directory does not contain a valid build tree:
#8 97.04 npm ERR! /app/node_modules/mediasoup/worker/out/Release
#8 97.04 npm ERR! make: *** [Makefile:128: mediasoup-worker] Error 1
#8 97.05 
#8 97.05 npm ERR! A complete log of this run can be found in:
#8 97.05 npm ERR!     /root/.npm/_logs/2022-07-04T08_44_09_381Z-debug-0.log
#8 ERROR: executor failed running [/bin/sh -c npm ci]: exit code: 1
#4 [1/5] FROM docker.io/library/node:16.15.0-bullseye-slim@sha256:fab24688ec5c8cb3db76ce89f32ae9ab979bf12912789db28dd0436f6f27bb07
------
 > [5/5] RUN npm ci:
#8 97.04 npm ERR! [729/992] Compiling C object subprojects/libuv-v1.42.0/libuv.a.p/src_unix_core.c.o
#8 97.04 npm ERR! ninja: build stopped: subcommand failed.
#8 97.04 npm ERR! make: Leaving directory '/app/node_modules/mediasoup/worker'
#8 97.04 npm ERR! WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#8 97.04 npm ERR! Directory does not contain a valid build tree:
#8 97.04 npm ERR! /app/node_modules/mediasoup/worker/out/Release
#8 97.04 npm ERR! make: *** [Makefile:128: mediasoup-worker] Error 1
#8 97.05 
#8 97.05 npm ERR! A complete log of this run can be found in:
#8 97.05 npm ERR!     /root/.npm/_logs/2022-07-04T08_44_09_381Z-debug-0.log
------
error: failed to solve: executor failed running [/bin/sh -c npm ci]: exit code: 1
Error: buildx failed with: error: failed to solve: executor failed running [/bin/sh -c npm ci]: exit code: 1
```

**Comment by ibc** [comment]
As said above, use the forum for asking about problems.

**Comment by mizoRC** [comment]
Maybe related with this commit? https://github.com/versatica/mediasoup/commit/e8d6d53f943080934614c2371065a3fd5a3ed91f

When will this patch be published on npm?

**Comment by the-reality-engineer** [comment]
Others are suddenly seeing this issue as well. https://mediasoup.discourse.group/t/mediasoup-build-fails/4345

**Comment by mizoRC** [comment]
Fixed on 3.10.2.
